code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/AccountAccess.java

```diff
@@ -0,0 +1,55 @@
+package io.cattle.platform.iaas.api.auth;
+
+import io.cattle.platform.api.auth.ExternalId;
+import io.cattle.platform.core.model.Account;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class AccountAccess {
+
+    Account account;
+    Account project;
+    Set<ExternalId> externalIds;
+
+    public AccountAccess() {
+    }
+
+    public AccountAccess(Account account, Set<ExternalId> externalIdSet) {
+        this.account = account;
+        this.project = account;
+        this.externalIds = externalIdSet;
+        if (account != null){
```

Comment:
James, could you explain this piece of logic? In what scenario account can be null, and why not initialize externalIds to an empty set in the field declaration, and then in the constructor just do:

if (externalIdSet != null) {
this.externalIds = externalIdSet
}